skimage/morphology/_flood_fill.py

```diff
@@ -0,0 +1,290 @@
+        equal to `connectivity` are considered neighbors. Ignored if
+        `selem` is not None.
+    tolerance : float or int, optional
+        If None (default), adjacent values must be strictly equal to the
+        initial value of `image` at `seed_point`.  This is fastest.  If a value
+        is given, a comparison will be done at every point and if within
+        tolerance of the initial value will also be filled (inclusive).
+
+    Returns
+    -------
+    mask : ndarray
+        A Boolean array with the same shape as `image` is returned, with values
+        equal to 1 for areas connected to and equal (or within tolerance of)
+        the seed point.  All other values are zero.
+
+    Notes
+    -----
+    The conceptual analogy of this operation is the 'paint bucket' tool in many
+    raster graphics programs.  This function returns just the mask
+    representing the fill.
+
+    If indices are desired rather than masks for memory reasons, the user can
+    simply run `numpy.nonzero` on the result, save the indices, and discard
+    this mask.
+
+    Examples
+    --------
+    >>> from skimage.morphology import flood
+    >>> image = np.zeros((4, 7), dtype=int)
+    >>> image[1:3, 1:3] = 1
+    >>> image[3, 0] = 1
+    >>> image[1:3, 4:6] = 2
+    >>> image[3, 6] = 3
+    >>> image
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [1, 0, 0, 0, 0, 0, 3]])
+
+    Fill connected ones with 5, with full connectivity (diagonals included):
+
+    >>> mask = flood(image, (1, 1))
+    >>> image[mask] = 5
+    >>> image
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [5, 0, 0, 0, 0, 0, 3]])
+
+    Fill connected fives with 1, with only cardinal direction connectivity:
+
+    >>> mask = flood(image, (1, 1), connectivity=1)
+    >>> image[mask] = 1
+    >>> image
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [5, 0, 0, 0, 0, 0, 3]])
+
+    Fill with a tolerance:
+
+    >>> mask = flood(image, (0, 0), tolerance=1)
+    >>> image[mask] = 5
+    >>> image
+    array([[5, 5, 5, 5, 5, 5, 5],
+           [5, 5, 5, 5, 2, 2, 5],
+           [5, 5, 5, 5, 2, 2, 5],
+           [5, 5, 5, 5, 5, 5, 3]])
+    """
+    # Correct start point in ravelled image - only copy if non-contiguous
+    image = np.asarray(image)
+    if image.flags.f_contiguous is True:
+        order = 'F'
+    elif image.flags.c_contiguous is True:
+        order = 'C'
+    else:
+        warnings.warn('Non-contiguous array passed as `image`; this will be '
+                      'converted to a contiguous array as a copy.')
+        image = np.ascontiguousarray(image)
+        order = 'C'
+
+    seed_value = image[seed_point]
+
+    # Shortcut for rank zero
+    if image.size == 0:
+        return np.array([], dtype=np.uint8)
```

Comment:
The docstring promises that this function returns "A Boolean array with the same shape as `image`".

```suggestion
        return np.zeros(image.shape, dtype=np.uint8)
```

Reply:
@lagru shouldn't the dtype be `bool`? But your suggestion about the shape is spot-on — even if the user passes an array of shape (0, 0), we want to preserve that dimensionality. It might be useful for stacking things later, for example.

Reply:
@jni Yeah, good catch! I totally overlooked that this function returns a boolean array.

Reply:
@lagru You literally mentioned it in your comment! :joy:

Reply:
Yeah, I got to that part of the code after suggesting this change. :sweat_smile: